org.elasticsearch.xpack.esql.qa.single_node.EsqlClientYamlIT test {p0=/61_enrich_ip/IP strings} fails

### CI Link

https://gradle-enterprise.elastic.co/s/aokqdwvbv5hd6/tests/task/:x-pack:plugin:esql:qa:server:single-node:yamlRestTest/details/org.elasticsearch.xpack.esql.qa.single_node.EsqlClientYamlIT/test%20%7Bp0=%2F61_enrich_ip%2FIP%20strings%7D?top-execution=1

### Repro line

./gradlew ':x-pack:plugin:esql:qa:server:single-node:yamlRestTest' --tests "org.elasticsearch.xpack.esql.qa.single_node.EsqlClientYamlIT.test {p0=/61_enrich_ip/IP strings}" -Dtests.seed=485FE5D92BABAE7C -Dtests.locale=ar-TN -Dtests.timezone=Asia/Anadyr

### Does it reproduce?

No

### Applicable branches

main

### Failure history

Another sighting here:
  https://gradle-enterprise.elastic.co/s/qi5sm3nivjrly/tests/:x-pack:plugin:esql:qa:server:single-node:yamlRestTest/org.elasticsearch.xpack.esql.qa.single_node.EsqlClientYamlIT/test%20%7Bp0=%2F61_enrich_ip%2FIP%20strings%7D

### Failure excerpt

```
java.lang.AssertionError: Failure at [/61_enrich_ip:83]: Expected a list containing |  
-- | --
0: a list containing |  
0: "10.100.0.21" |  
1: "10.101.0.107" |  
1: a list containing |  
0: expected "Production" but was "QA" |  
1: expected "QA" but was "Production" |  
2: a list containing |  
0: expected "OPS" but was "Engineering" |  
1: expected "Engineering" but was "OPS" |  
3: "sending messages"
```

--- a/x-pack/plugin/esql/qa/server/single-node/build.gradle
+++ b/x-pack/plugin/esql/qa/server/single-node/build.gradle
@@ -7,7 +7,7 @@ dependencies {
 
 restResources {
   restApi {
-    include '_common', 'bulk', 'indices', 'esql', 'xpack', 'enrich'
+    include '_common', 'bulk', 'indices', 'esql', 'xpack', 'enrich', 'cluster'
   }
 }
 

--- a/x-pack/plugin/esql/qa/server/single-node/src/yamlRestTest/resources/rest-api-spec/test/61_enrich_ip.yml
+++ b/x-pack/plugin/esql/qa/server/single-node/src/yamlRestTest/resources/rest-api-spec/test/61_enrich_ip.yml
@@ -4,6 +4,9 @@ setup:
       indices.create:
         index: networks
         body:
+          settings:
+            index.number_of_shards: 1
+            index.routing.rebalance.enable: "none"
           mappings:
             properties:
               range:
@@ -22,6 +25,10 @@ setup:
           - { "range": "10.100.0.0/16", "name": "Production", "department": "OPS" }
           - { "index": { } }
           - { "range": "10.101.0.0/16", "name": "QA", "department": "Engineering" }
+  - do:
+      cluster.health:
+        wait_for_no_initializing_shards: true
+        wait_for_events: languid
 
   - do:
       enrich.put_policy:
@@ -64,9 +71,6 @@ setup:
           - { "@timestamp": "2023-06-24", "ip": "13.101.0.114", "message": "authentication failed" }
 ---
 "IP strings":
-  - skip:
-      version: all
-      reason: "AwaitsFix https://github.com/elastic/elasticsearch/issues/99807"
 
   - do:
       esql.query:

